setuptools_scm/version.py

```diff
@@ -144,6 +144,20 @@ def get_local_node_and_date(version):
         return version.format_choice("+{node}", "+{node}.d{time:%Y%m%d}")
 
 
+def get_local_node_and_timestamp(version, fmt='%Y%m%d%H%M%S'):
+    if version.exact or version.node is None:
+        return version.format_choice("",
+                                     "+d{time:"
+                                     + "{fmt}".format(fmt=fmt)
```

Comment:
this should use nested replacements as per https://www.python.org/dev/peps/pep-3101/#format-specifiers

this needs an api addition however

Reply:
a follow-up issue was created